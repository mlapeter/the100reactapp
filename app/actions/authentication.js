export const FETCH_TOKEN = "FETCH_TOKEN";
export const FETCH_TOKEN_RESULT = "FETCH_TOKEN_RESULT";
export const FETCH_TOKEN_ERROR = "FETCH_TOKEN_ERROR";

export const DECODE_TOKEN = "DECODE_TOKEN";
export const DECODE_TOKEN_RESULT = "DECODE_TOKEN_RESULT";
export const DECODE_TOKEN_ERROR = "DECODE_TOKEN_ERROR";
export const REMOVE_TOKEN = "REMOVE_TOKEN";
export const REMOVE_TOKEN_ERROR = "REMOVE_TOKEN_ERROR";
export const SET_FIREBASE_TOKEN = "SET_FIREBASE_TOKEN";
export const RESET_PASSWORD = "RESET_PASSWORD";
export const RESET_PASSWORD_RESULT = "RESET_PASSWORD_RESULT";
export const RESET_PASSWORD_ERROR = "RESET_PASSWORD_ERROR";

export const fetchToken = (username, password, temp_auth_token) => ({
  type: FETCH_TOKEN,
  username,
  password,
  temp_auth_token
});

export const decodeToken = token => ({
  type: DECODE_TOKEN,
  token
});

export const removeToken = () => ({
  type: REMOVE_TOKEN
});

export const setFirebaseToken = token => ({
  type: SET_FIREBASE_TOKEN,
  token
});

export const resetPassword = resetPasswordEmail => ({
  type: RESET_PASSWORD,
  resetPasswordEmail
});
