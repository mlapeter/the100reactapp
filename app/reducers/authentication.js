import {
  FETCH_TOKEN,
  FETCH_TOKEN_RESULT,
  FETCH_TOKEN_ERROR,
  DECODE_TOKEN,
  DECODE_TOKEN_RESULT,
  DECODE_TOKEN_ERROR,
  REMOVE_TOKEN,
  REMOVE_TOKEN_ERROR,
  SET_FIREBASE_TOKEN
} from "../actions/authentication";

const initialState = {
  appLoading: true,
  isLoading: false,
  isAuthed: false,
  user: {},
  token: "",
  firebaseToken: ""
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_TOKEN:
      return {
        ...state,
        username: action.username,
        password: action.password,
        isLoading: true
      };
    case FETCH_TOKEN_RESULT:
      return {
        ...state,
        token: action.token,
        firebaseToken: action.firebaseToken
      };
    case FETCH_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false,
        token: "",
        firebaseToken: ""
      };
    case DECODE_TOKEN:
      return {
        ...state,
        token: action.token,
        isLoading: true
      };
    case DECODE_TOKEN_RESULT:
      return {
        ...state,
        user: action.result,
        isLoading: false,
        isAuthed: true
      };
    case DECODE_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false
      };
    case REMOVE_TOKEN:
      return {
        ...state,
        isAuthed: false,
        isLoading: false,
        user: {},
        token: "",
        firebaseToken: ""
      };
    case REMOVE_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false,
        user: {},
        token: ""
      };
    case SET_FIREBASE_TOKEN:
      return {
        ...state,
        firebaseToken: action.token
      };
    default:
      return state;
  }
};
