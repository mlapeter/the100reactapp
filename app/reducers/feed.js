import {
  FETCH_FEED,
  FETCH_FEED_RESULT,
  FETCH_FEED_ERROR,
  FETCH_FEED_NO_DATA,
  LOAD_MORE_FEED_ITEMS,
  LOAD_MORE_FEED_ITEMS_RESULT,
  CHANGE_FEED_PAGE
} from "../actions/feed";

const initialState = {
  isLoading: false,
  feedItems: [],
  feedPage: 1,
  moreFeedItemsAvailable: true
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_FEED:
      return {
        ...state,
        isLoading: true,
        moreFeedItemsAvailable: true
      };
    case FETCH_FEED_RESULT:
      return {
        ...state,
        feedItems: action.result,
        isLoading: false
      };
    case FETCH_FEED_ERROR:
      return {
        ...state,
        error: action.error,
        isLoading: false
      };
    case FETCH_FEED_NO_DATA:
      return {
        ...state,
        isLoading: false,
        moreFeedItemsAvailable: false
      };
    case LOAD_MORE_FEED_ITEMS:
      return {
        ...state,
        isLoading: true
      };
    case LOAD_MORE_FEED_ITEMS_RESULT:
      return {
        ...state,
        feedItems: [...state.feedItems, ...action.result],
        isLoading: false
      };

    case CHANGE_FEED_PAGE:
      return {
        ...state,
        feedPage: action.page
      };
    default:
      return state;
  }
};
