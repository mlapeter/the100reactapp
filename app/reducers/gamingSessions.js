import Environment from "../config/environment";

import {
  CREATE_GAMING_SESSION,
  CREATE_GAMING_SESSION_RESULT,
  CREATE_GAMING_SESSION_ERROR,
  EDIT_GAMING_SESSION,
  EDIT_GAMING_SESSION_RESULT,
  EDIT_GAMING_SESSION_ERROR,
  DELETE_GAMING_SESSION,
  DELETE_GAMING_SESSION_RESULT,
  DELETE_GAMING_SESSION_ERROR,
  FETCH_GAMING_SESSION,
  FETCH_GAMING_SESSION_RESULT,
  FETCH_GAMING_SESSION_ERROR,
  FETCH_GAMING_SESSIONS,
  FETCH_GAMING_SESSIONS_RESULT,
  FETCH_GAMING_SESSIONS_ERROR,
  FETCH_GAMING_SESSIONS_NO_DATA,
  FETCH_GAMING_SESSIONS_NO_SEARCH_RESULTS,
  REFRESH_GAMING_SESSIONS,
  LOAD_MORE_GAMING_SESSIONS,
  LOAD_MORE_GAMING_SESSIONS_RESULT,
  FETCH_MY_GAMING_SESSIONS,
  FETCH_MY_GAMING_SESSIONS_RESULT,
  FETCH_MY_GAMING_SESSIONS_ERROR,
  FETCH_MY_GAMING_SESSIONS_NO_DATA,
  CLEAR_MY_GAMING_SESSIONS,
  REFRESH_MY_GAMING_SESSIONS,
  LOAD_MORE_MY_GAMING_SESSIONS,
  LOAD_MORE_MY_GAMING_SESSIONS_RESULT,
  FETCH_GROUP_GAMING_SESSIONS,
  FETCH_GROUP_GAMING_SESSIONS_RESULT,
  FETCH_GROUP_GAMING_SESSIONS_ERROR,
  FETCH_GROUP_GAMING_SESSIONS_NO_DATA,
  CLEAR_GROUP_GAMING_SESSIONS,
  REFRESH_GROUP_GAMING_SESSIONS,
  LOAD_MORE_GROUP_GAMING_SESSIONS,
  LOAD_MORE_GROUP_GAMING_SESSIONS_RESULT,
  FETCH_RECENT_GAMING_SESSIONS,
  FETCH_RECENT_GAMING_SESSIONS_RESULT,
  FETCH_RECENT_GAMING_SESSIONS_ERROR,
  FETCH_RECENT_GAMING_SESSIONS_NO_DATA,
  CLEAR_RECENT_GAMING_SESSIONS,
  REFRESH_RECENT_GAMING_SESSIONS,
  LOAD_MORE_RECENT_GAMING_SESSIONS,
  LOAD_MORE_RECENT_GAMING_SESSIONS_RESULT
} from "../actions/gamingSessions";

const initialState = {
  endpoint:
    Environment["API_BASE_URL"] +
    Environment["API_VERSION"] +
    "gaming_sessions",
  refreshing: false,
  isCreating: false,
  gameCreated: false,
  isLoading: false,
  gameEdited: false,
  gamingSessionId: null,
  gamingSession: {},
  gamingSessionLoading: true,

  moreGamingSessionsAvailable: true,
  moreMyGamingSessionsAvailable: true,
  moreGroupGamingSessionsAvailable: true,
  moreRecentGamingSessionsAvailable: true,

  gamingSessionsRefreshing: false,
  myGamingSessionsRefreshing: false,
  groupGamingSessionsRefreshing: false,
  recentGamingSessionsRefreshing: false,

  gamingSessions: [],
  myGamingSessions: [],
  groupGamingSessions: [],
  recentGamingSessions: []
};

export default (state = initialState, action) => {
  switch (action.type) {
    case CREATE_GAMING_SESSION:
      return {
        ...state,
        isCreating: true,
        gamingSession: action.gamingSession
      };
    case CREATE_GAMING_SESSION_RESULT:
      return {
        ...state,
        isCreating: false,
        gameCreated: true,
        successAt: new Date(),
        gamingSession: action.result
      };
    case CREATE_GAMING_SESSION_ERROR:
      return {
        ...state,
        isCreating: false,
        gameCreated: false,
        error: action.error,
        errorAt: new Date()
      };
    case EDIT_GAMING_SESSION:
      return {
        ...state,
        gameEdited: false,
        gameDeleted: false,
        isLoading: true,
        gamingSession: action.gamingSession
      };
    case EDIT_GAMING_SESSION_RESULT:
      return {
        ...state,
        isLoading: false,
        gameEdited: true,
        successAt: new Date(),
        gamingSession: action.result
      };
    case EDIT_GAMING_SESSION_ERROR:
      return {
        ...state,
        isLoading: false,
        gameEdited: false,
        error: action.error,
        errorAt: new Date()
      };
    case DELETE_GAMING_SESSION:
      return {
        ...state,
        gameEdited: false,
        gameDeleted: false,
        isLoading: true,
        deleteGamingSessionId: action.deleteGamingSessionId
      };
    case DELETE_GAMING_SESSION_RESULT:
      return {
        ...state,
        isLoading: false,
        gameDeleted: true,
        successAt: new Date(),
        gamingSession: action.result
      };
    case DELETE_GAMING_SESSION_ERROR:
      return {
        ...state,
        isLoading: false,
        gameDeleted: false,
        error: action.error,
        errorAt: new Date()
      };
    case FETCH_GAMING_SESSION:
      return {
        ...state,
        gamingSessionLoading: true,
        gamingSessionId: action.gamingSessionId
      };
    case FETCH_GAMING_SESSION_RESULT:
      return {
        ...state,
        gamingSession: action.result,
        gamingSessionLoading: false
      };
    case FETCH_GAMING_SESSION_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        gamingSessionLoading: true
      };
    case FETCH_GAMING_SESSIONS:
      return {
        ...state,
        moreGamingSessionsAvailable: true,
        gamingSessionsRefreshing: true,
        endpoint: action.endpoint
      };
    case FETCH_GAMING_SESSIONS_RESULT:
      return {
        ...state,
        gamingSessions: action.result,
        gamingSessionsRefreshing: false
      };
    case FETCH_GAMING_SESSIONS_ERROR:
      return {
        ...state,
        error: action.error,
        gamingSessionsRefreshing: false,
        moreGamingSessionsAvailable: false
      };
    case FETCH_GAMING_SESSIONS_NO_DATA:
      return {
        ...state,
        moreGamingSessionsAvailable: false,
        gamingSessionsRefreshing: false
      };
    case FETCH_GAMING_SESSIONS_NO_SEARCH_RESULTS:
      return {
        ...state,
        gamingSessions: [],
        moreGamingSessionsAvailable: false,
        gamingSessionsRefreshing: false
      };

    case REFRESH_GAMING_SESSIONS:
      return {
        ...state,
        moreGamingSessionsAvailable: true,
        gamingSessionsRefreshing: true,
        endpoint: action.endpoint
      };
    case LOAD_MORE_GAMING_SESSIONS:
      return {
        ...state,
        gamingSessionsRefreshing: true,
        endpoint: action.endpoint
      };
    case LOAD_MORE_GAMING_SESSIONS_RESULT:
      return {
        ...state,
        gamingSessions: [...state.gamingSessions, ...action.result],
        gamingSessionsRefreshing: false
      };
    case FETCH_MY_GAMING_SESSIONS:
      return {
        ...state,
        myGamingSessionsRefreshing: true,
        moreMyGamingSessionsAvailable: true
      };
    case FETCH_MY_GAMING_SESSIONS_RESULT:
      return {
        ...state,
        myGamingSessions: action.result,
        myGamingSessionsRefreshing: false
      };
    case FETCH_MY_GAMING_SESSIONS_ERROR:
      return {
        ...state,
        error: action.error,
        myGamingSessionsRefreshing: false
      };
    case FETCH_MY_GAMING_SESSIONS_NO_DATA:
      return {
        ...state,
        moreMyGamingSessionsAvailable: false,
        myGamingSessionsRefreshing: false
      };
    case CLEAR_MY_GAMING_SESSIONS:
      return {
        ...state,
        myGamingSessions: [],
        moreMyGamingSessionsAvailable: true,
        endpoint: action.endpoint
      };
    case REFRESH_MY_GAMING_SESSIONS:
      return {
        ...state,
        moreMyGamingSessionsAvailable: true,
        myGamingSessionsRefreshing: true,
        endpoint: action.endpoint
      };
    case LOAD_MORE_MY_GAMING_SESSIONS:
      return {
        ...state,
        myGamingSessionsRefreshing: true,
        endpoint: action.endpoint
      };
    case LOAD_MORE_MY_GAMING_SESSIONS_RESULT:
      return {
        ...state,
        myGamingSessions: [...state.myGamingSessions, ...action.result],
        myGamingSessionsRefreshing: false
      };
    case FETCH_GROUP_GAMING_SESSIONS:
      return {
        ...state,
        groupGamingSessionsLoading: true
      };
    case FETCH_GROUP_GAMING_SESSIONS_RESULT:
      return {
        ...state,
        moreGroupGamingSessionsAvailable: true,
        groupGamingSessions: action.result,
        groupGamingSessionsRefreshing: false
      };
    case FETCH_GROUP_GAMING_SESSIONS_ERROR:
      return {
        ...state,
        error: action.error,
        groupGamingSessionsRefreshing: false
      };
    case FETCH_GROUP_GAMING_SESSIONS_NO_DATA:
      return {
        ...state,
        moreGroupGamingSessionsAvailable: false,
        groupGamingSessionsRefreshing: false
      };
    case CLEAR_GROUP_GAMING_SESSIONS:
      return {
        ...state,
        groupGamingSessions: [],
        moreGroupGamingSessionsAvailable: true,
        endpoint: action.endpoint
      };
    case REFRESH_GROUP_GAMING_SESSIONS:
      return {
        ...state,
        moreGroupGamingSessionsAvailable: true,
        groupGamingSessionsRefreshing: true,
        endpoint: action.endpoint
      };
    case LOAD_MORE_GROUP_GAMING_SESSIONS:
      return {
        ...state,
        groupGamingSessionsRefreshing: true,
        endpoint: action.endpoint
      };
    case LOAD_MORE_GROUP_GAMING_SESSIONS_RESULT:
      return {
        ...state,
        groupGamingSessions: [...state.groupGamingSessions, ...action.result],
        groupGamingSessionsRefreshing: false
      };
    case FETCH_RECENT_GAMING_SESSIONS:
      return {
        ...state,
        recentGamingSessionsRefreshing: true,
        moreRecentGamingSessionsAvailable: true
      };
    case FETCH_RECENT_GAMING_SESSIONS_RESULT:
      return {
        ...state,
        recentGamingSessions: action.result,
        recentGamingSessionsRefreshing: false
      };
    case FETCH_RECENT_GAMING_SESSIONS_ERROR:
      return {
        ...state,
        error: action.error,
        recentGamingSessionsRefreshing: false
      };
    case FETCH_RECENT_GAMING_SESSIONS_NO_DATA:
      return {
        ...state,
        moreRecentGamingSessionsAvailable: false,
        recentGamingSessionsRefreshing: false
      };
    case CLEAR_RECENT_GAMING_SESSIONS:
      return {
        ...state,
        recentGamingSessions: [],
        moreRecentGamingSessionsAvailable: true,
        endpoint: action.endpoint
      };
    case REFRESH_RECENT_GAMING_SESSIONS:
      return {
        ...state,
        moreRecentGamingSessionsAvailable: true,
        recentGamingSessionsRefreshing: true,
        endpoint: action.endpoint
      };
    case LOAD_MORE_RECENT_GAMING_SESSIONS:
      return {
        ...state,
        recentGamingSessionsRefreshing: true,
        endpoint: action.endpoint
      };
    case LOAD_MORE_RECENT_GAMING_SESSIONS_RESULT:
      return {
        ...state,
        recentGamingSessions: [...state.recentGamingSessions, ...action.result],
        recentGamingSessionsRefreshing: false
      };
    default:
      return state;
  }
};
