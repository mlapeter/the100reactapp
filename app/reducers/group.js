import {
  FETCH_GROUP,
  FETCH_GROUP_RESULT,
  FETCH_GROUP_ERROR,
  FETCH_GROUP_EMPTY,
  CHANGE_GROUP
} from "../actions/group";

const initialState = {
  isLoading: true,
  group: null,
  selectedGroupId: null
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_GROUP:
      return {
        ...state,
        isLoading: true,
        selectedGroupId: action.groupId
      };
    case FETCH_GROUP_RESULT:
      return {
        ...state,
        group: action.result,
        isLoading: false
      };
    case FETCH_GROUP_ERROR:
      return {
        ...state,
        error: action.error,
        isLoading: false
      };
    case FETCH_GROUP_EMPTY:
      return {
        ...state,
        isLoading: false
      };
    case CHANGE_GROUP:
      return {
        ...state,
        selectedGroupId: action.groupId
      };
    default:
      return state;
  }
};
