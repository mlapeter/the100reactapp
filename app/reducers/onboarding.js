import {
  SET_PLATFORM,
  SET_GAMERTAG,
  SET_GAME,
  SET_PROFILE_INFO,
  CREATE_USER,
  CREATE_USER_ERROR
} from "../actions/onboarding";

const initialState = {
  age: "",
  email: "",
  gamertag: "",
  password: "",
  platform: "",
  timezone: "",
  play_schedule: "",
  play_style: "",
  group: "",
  primary_game_id: "",
  tos_privacy_agreement: false
};

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_PLATFORM:
      return {
        ...state,
        platform: action.platform
      };
      break;
    case SET_GAMERTAG:
      return {
        ...state,
        gamertag: action.gamertag
      };
      break;
    case SET_GAME:
      return {
        ...state,
        primary_game_id: action.gameId
      };
      break;
    case SET_PROFILE_INFO:
      return {
        ...state,
        age: action.age,
        play_schedule: action.playSchedule,
        play_style: action.playStyle,
        time_zone: action.timezone,
        group: action.group
      };
      break;
    case CREATE_USER:
      return {
        ...state,
        email: action.email,
        password: action.password,
        tos_privacy_agreement: action.tos_privacy_agreement
      };
      break;
    case CREATE_USER_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        // platform: "",
        // gamertag: "",
        // timezone: "",
        // play_style: "",
        // age: "",
        // play_schedule: "",
        // group: "",
        // email: "",
        password: ""
      };
    default:
      return state;
  }
};
