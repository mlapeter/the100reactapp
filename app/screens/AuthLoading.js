import React from "react";
import {
  ActivityIndicator,
  AsyncStorage,
  StatusBar,
  StyleSheet,
  View
} from "react-native";
import { Font } from "expo";
import { connect } from "react-redux";
import { connectAlert } from "../components/Alert";
import { decodeToken, setFirebaseToken } from "../actions/authentication";
import { changeSelectedGroupId } from "../actions/group";

import { colors, fontSizes } from "../styles";
import PreSplash from "../components/PreSplash/PreSplash";

class AuthLoading extends React.Component {
  constructor(props) {
    super(props);
    this.bootstrap();
  }

  bootstrap = () => {
    console.log("Starting App");
    Font.loadAsync({
      Lato: require("../../app/assets/fonts/Lato-Bold.ttf"),
      Nunito: require("../../app/assets/fonts/Nunito-Bold.ttf")
    }).then(result => {
      AsyncStorage.getItem("fb_token").then(token => {
        this.props.dispatch(setFirebaseToken(token));
      });
      AsyncStorage.getItem("id_token").then(token => {
        this.props.dispatch(decodeToken(token));

        // setTimeout(() => {
        //   if (
        //     this.props.authentication.isAuthed === true &&
        //     this.props.users.currentUser.gamertag != null
        //   ) {
        //     this.props.navigation.navigate("App");
        //   } else {
        //     this.props.navigation.navigate("Auth");
        //   }
        // }, 3000);

        setTimeout(() => {
          if (
            !this.props.users.currentUser ||
            this.props.users.currentUser.gamertag == null ||
            this.props.authentication.isAuthed !== true
          ) {
            this.props.navigation.navigate("Auth");
          } else {
            this.props.navigation.navigate("App");
          }
        }, 3000);
      });
    });
  };

  // Fetch the token from storage then navigate to our appropriate place
  // _bootstrapAsync = async () => {
  //   const userToken = await AsyncStorage.getItem("userToken");
  //
  //   // This will switch to the App screen or Auth screen and this loading
  //   // screen will be unmounted and thrown away.
  //   this.props.navigation.navigate(userToken ? "App" : "Auth");
  // };

  // Render any loading content that you like here
  render() {
    return (
      <View style={styles.container}>
        <PreSplash />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    padding: 5,
    // paddingTop: 25,
    flex: 1,
    flexDirection: "column",
    justifyContent: "center",
    backgroundColor: colors.white
  }
});

const mapStateToProps = state => {
  const authentication = state.authentication;
  const users = state.users;
  // const onAuthChange = state.authentication.onAuthChange(token);

  return {
    authentication,
    users,
    authenticationError: state.authentication.error
  };
};

export default connect(mapStateToProps)(connectAlert(AuthLoading));
