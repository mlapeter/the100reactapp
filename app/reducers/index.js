import { combineReducers } from "redux";

import authentication from "./authentication";
import search from "./search";
import notifications from "./notifications";
import users from "./users";
import group from "./group";
import gamingSessions from "./gamingSessions";
import onboarding from "./onboarding";
import conversations from "./conversations";
import feed from "./feed";

export default combineReducers({
  authentication,
  search,
  notifications,
  users,
  group,
  gamingSessions,
  onboarding,
  conversations,
  feed
});
