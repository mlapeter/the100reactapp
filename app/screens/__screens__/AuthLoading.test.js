import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";

import AuthLoading from "../AuthLoading";
import { initialState } from "../../reducers/authentication";

const mockStore = configureStore([]);

it("successfully renders", () => {
  // const navigation = { state: { params: {} } };
  const initialState = { authentication: initialState };
  const rendered = shallow(<AuthLoading />, {
    context: { store: mockStore(initialState) }
  });
  expect(rendered.dive()).toMatchSnapshot();
});
