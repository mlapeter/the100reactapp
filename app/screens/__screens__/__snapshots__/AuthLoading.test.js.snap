// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`successfully renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConnectedAlert
    dispatch={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alert": undefined,
      "alertWithType": undefined,
      "authentication": undefined,
      "dispatch": [Function],
      "users": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alert": undefined,
        "alertWithType": undefined,
        "authentication": undefined,
        "dispatch": [Function],
        "users": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
